src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -312,12 +318,104 @@ private static void printLocales() {
             // print columns of 8
             if ((i + 1) % 8 == 0) {
                 ostream.println();
-                ostream.print(INDENT + INDENT);
+                ostream.print(TWOINDENT);
+            }
+        }
+        ostream.print("\n");
+    }
+
+    private static void printSecuritySettings(String arg) {
+        if (arg.toLowerCase(Locale.ROOT).equals("properties")) {
+            printSecurityProperties();
+        } else if(arg.toLowerCase(Locale.ROOT).equals("providers")) {
+            printSecurityProviderConfig();
+        } else if(arg.toLowerCase(Locale.ROOT).equals("tls")) {
+            printSecurityTLSConfig();
+        } else {
+            printSecurityProperties();
+            printSecurityProviderConfig();
+            printSecurityTLSConfig();
+        }
+    }
+
+    private static void printSecurityProperties() {
+        ostream.println(INDENT + "Security properties:");
+        Properties p = SharedSecrets.getJavaSecurityPropertiesAccess().getInitialProperties();
+        for (String key : p.stringPropertyNames().stream().sorted().toList()) {
+            String val = p.getProperty(key);
+            if (val.contains(",") && val.length() > 60) {
+                // split lines longer than 60 chars which have multiple values
+                ostream.println(TWOINDENT + key + "=");
+                List.of(val.split(",")).forEach(s -> ostream.println(THREEINDENT + s.trim()));
+            } else {
+                ostream.println(TWOINDENT + key + "=" + val);
+            }
+        }
+        ostream.print("\n");
+    }
+
+    private static void printSecurityTLSConfig() {
+        SSLSocket ssls;
+        try {
+            ssls = (SSLSocket)
+                    SSLContext.getDefault().getSocketFactory().createSocket();
+        } catch (IOException | NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
```

Comment:
That's pretty rude of the runtime; perhaps `throw InternalError("Unable to create secure socket")`.
It would give some clue that doesn't look like a coding bug.